components/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,481 @@
+        const res = Utils.applyHotkeyMarkdown(e);
+
+        setEditText(res.message);
+        if (textboxRef.current) {
+            Utils.setSelectionRange(
+                textboxRef.current.getInputBox(),
+                res.selectionStart,
+                res.selectionEnd,
+            );
+        }
+    };
+
+    const handleRefocusAndExit = (refocusId: string|null) => {
+        if (refocusId) {
+            setTimeout(() => {
+                const element = document.getElementById(refocusId);
+                if (element) {
+                    element.focus();
+                }
+            });
+        }
+
+        setCaretPosition(0);
+        setPostError(null);
+        setErrorClass('');
+        setShowEmojiPicker(false);
+        setRenderScrollbar(false);
+    };
+
+    const handleHide = (doRefocus = true) => {
+        setEditText(editingPost.post?.message || '');
+        actions.unsetEditingPost();
+        handleRefocusAndExit(doRefocus && editingPost.refocusId ? editingPost.refocusId : null);
+    };
+
+    const handleEdit = async () => {
+        if (!editingPost.post || isSaveDisabled()) {
+            return;
+        }
+
+        const updatedPost = {
+            message: editText,
+            id: editingPost.postId,
+            channel_id: editingPost.post.channel_id,
+        };
+
+        if (postError) {
+            setErrorClass('animation--highlight');
+            setTimeout(() => setErrorClass(''), Constants.ANIMATION_TIMEOUT);
+            return;
+        }
+
+        if (updatedPost.message === (editingPost.post?.message_source || editingPost.post?.message)) {
+            handleHide();
+            return;
+        }
+
+        const hasAttachment = Boolean(
+            editingPost.post?.file_ids && editingPost.post?.file_ids.length > 0,
+        );
+        if (updatedPost.message.trim().length === 0 && !hasAttachment) {
+            handleHide(false);
+
+            const deletePostModalData = {
+                ModalId: ModalIdentifiers.DELETE_POST,
+                dialogType: DeletePostModal,
+                dialogProps: {
+                    post: editingPost.post,
+                    isRHS: editingPost.isRHS,
+                },
+            };
+
+            actions.openModal(deletePostModalData);
+            return;
+        }
+
+        actions.addMessageIntoHistory(updatedPost.message);
+
+        const data = await actions.editPost(updatedPost as Post);
+
+        if (data) {
+            window.scrollTo(0, 0);
+        }
+
+        handleHide();
+    };
+
+    const handleEditKeyPress = (e: React.KeyboardEvent) => {
+        const {ctrlSend, codeBlockOnCtrlEnter} = rest;
+
+        const {allowSending, ignoreKeyPress} = postMessageOnKeyPress(
+            e,
+            editText,
+            ctrlSend,
+            codeBlockOnCtrlEnter,
+            Date.now(),
+            0,
+            caretPosition,
+        ) as {allowSending: boolean; ignoreKeyPress?: boolean};
```

Comment:
Why is the `as` needed? Can we make it so it is not?

Reply:
it is not ...